modules/nf-core/gffread/main.nf

```diff
@@ -1,32 +1,59 @@
 process GFFREAD {
-    tag "$gff"
+    tag "$meta.id"
     label 'process_low'
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/gffread:0.12.1--h8b12597_0' :
-        'biocontainers/gffread:0.12.1--h8b12597_0' }"
+        'https://depot.galaxyproject.org/singularity/gffread:0.12.7--hdcf5f25_4' :
+        'biocontainers/gffread:0.12.7--hdcf5f25_4' }"
 
     input:
-    path gff
+    tuple val(meta), path(gff)
+    path fasta
 
     output:
-    path "*.gtf"        , emit: gtf         , optional: true
-    path "*.gff3"       , emit: gffread_gff , optional: true
-    path "versions.yml" , emit: versions
+    tuple val(meta), path("*.gtf")  , emit: gtf             , optional: true
+    tuple val(meta), path("*.gff3") , emit: gffread_gff     , optional: true
+    tuple val(meta), path("*.fasta"), emit: gffread_fasta   , optional: true
+    path "versions.yml"             , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args        = task.ext.args   ?: ''
-    def prefix      = task.ext.prefix ?: "${gff.baseName}"
-    def extension   = args.contains("-T") ? 'gtf' : 'gffread.gff3'
+    def args        = task.ext.args             ?: ''
+    def prefix      = task.ext.prefix           ?: "${meta.id}"
+    def extension   = args.contains("-T")       ? 'gtf' : ( ( ['-w', '-x', '-y' ].any { args.contains(it) } ) ? 'fasta' : 'gff3' )
+    def fasta_arg   = fasta                     ? "-g $fasta" : ''
+    def output_name = "${prefix}.${extension}"
+    def output      = extension == "fasta"      ? "$output_name" : "-o $output_name"
+    def args_sorted = args.replaceAll(/(.*)(-[wxy])(.*)/) { all, pre, param, post -> "$pre $post $param" }.trim()
```

Comment:
Can you explain what this is doing?

Reply:
Thank you @SPPearce 

A fasta file is produced when the user specifies '-w', '-x', or '-y' param with a file name. Otherwise, the output file name (both gtf and gff3) should be specified with '-o'.

This line moves '-w', '-x', or '-y', if present, to the end of the `args` string and then the file name is specified by `output`.

Reply:
Hmm, so it would normally be -w PATH, but you only need to specify -w here? 

I think this needs would benefit from comment in the code to explain what this is doing as it isn't obvious.

Reply:
If you can add a comment explaining this bit of code I think it is ready after that.

Reply:
Thank you @SPPearce 

I have added a comment now:

https://github.com/nf-core/modules/blob/dbdb6e7ac2015087d5e54dc2907c57bc7b205613/modules/nf-core/gffread/main.nf#L31